### PR TITLE
[AzureRG Deploy] Parsing the override parameter based on the type

### PR DESCRIPTION
**Pull request by ds-ms**

### CODE REVIEW

File: Tasks/AzureResourceGroupDeployment/Strings/resources.resjson/en-US/resources.resjson [hunk 1]
```diff
@@ -45,7 +45,7 @@
   "loc.messages.CreatedRG": "Resource Group created successfully.",
   "loc.messages.CreatingTemplateDeployment": "Creating deployment parameters.",
   "loc.messages.TemplateParsingFailed": "Ensure the Template file is valid. Task failed while parsing with following error: %s",
-  "loc.messages.ParametersFileFetchFailed": "Failed to download the parameters file. Error: %s",
+  "loc.messages.FileFetchFailed": "Failed to download the parameters file. Error: %s",
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/Strings/resources.resjson/en-US/resources.resjson]: The message still says failed to download the parameters file.

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts [hunk 2]
```diff
@@ -146,12 +146,21 @@ export class ResourceGroup {
...
-    private updateOverrideParameters(parameters: Object): Object {
+    private updateOverrideParameters(template: Object, parameters: Object): Object {
         tl.debug("Overriding Parameters..");
 
         var override = parameterParser(this.taskParameters.overrideParameters);
         for (var key in override) {
             tl.debug("Overriding key: " + key);
+            try {
+                if (template["parameters"][key]["type"] == "int") {
+                    override[key]["value"] = parseInt(override[key]["value"]);
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: What if the value is not a int

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts [hunk 3]
```diff
@@ -146,12 +146,21 @@ export class ResourceGroup {
...
+                }
+            } catch (error) {
+                // Adding parameter which isn't present in the template file
+                tl.debug(error.toString());
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: i think we should have a better message here. And shouldn't it be a warning?

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts [hunk 4]
```diff
@@ -146,12 +146,21 @@ export class ResourceGroup {
...
 
-    private updateOverrideParameters(parameters: Object): Object {
+    private updateOverrideParameters(template: Object, parameters: Object): Object {
         tl.debug("Overriding Parameters..");
 
         var override = parameterParser(this.taskParameters.overrideParameters);
         for (var key in override) {
             tl.debug("Overriding key: " + key);
+            try {
+                if (template["parameters"][key]["type"] == "int") {
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: Are there more variable types which might require special behavior? Can you think through all the variable types and see that our behavior is correct

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts (hunk 2 above)
Comment (ds-ms) [inline Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: If the value is not int, it'll be treated as a string

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts (hunk 3 above)
Comment (ds-ms) [inline Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: Sure

File: Tasks/AzureResourceGroupDeployment/Strings/resources.resjson/en-US/resources.resjson (hunk 1 above)
Comment (ds-ms) [inline Tasks/AzureResourceGroupDeployment/Strings/resources.resjson/en-US/resources.resjson]: Will change

File: Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts [hunk 5]
```diff
@@ -146,15 +146,41 @@ export class ResourceGroup {
...
+        var overrideParameters = parameterParser(this.taskParameters.overrideParameters);
+        for (var key in overrideParameters) {
             tl.debug("Overriding key: " + key);
-            parameters[key] = override[key];
-        }
+            try {
+                overrideParameters[key]["value"] = this.castToType(overrideParameters[key]["value"], template["parameters"][key]["type"]);
+            } catch (error) {
+                // Adding parameter which isn't present in the template file
+                if (error.name === "SyntaxError")
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/operations/ResourceGroup.ts]: Shouldn't you log the warning for all kinds of errors.

File: Tasks/AzureResourceGroupDeployment/Strings/resources.resjson/en-US/resources.resjson [hunk 6]
```diff
@@ -45,7 +45,7 @@
   "loc.messages.CreatedRG": "Resource Group created successfully.",
   "loc.messages.CreatingTemplateDeployment": "Creating deployment parameters.",
   "loc.messages.TemplateParsingFailed": "Ensure the Template file is valid. Task failed while parsing with following error: %s",
-  "loc.messages.ParametersFileFetchFailed": "Failed to download the parameters file. Error: %s",
+  "loc.messages.FileFetchFailed": "Failed to download the file. URL: %s Error: %s",
```
Comment (sachinma) [inline Tasks/AzureResourceGroupDeployment/Strings/resources.resjson/en-US/resources.resjson]: Before Error have a .

**Pull request merged**